packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/IrUtils.kt

```diff
@@ -434,6 +438,26 @@ fun IrBlockBuilder.createSafeCallConstruction(
     }
 }
 
+/**
+ * Using reflection to invoke the `arguments` attribute of the `IrSimpleType` to determine the enclosing type
+ * of the `RealmList`.
+ * This work around is needed since `IrSimpleType` became an abstract class in Kotlin 1.7 (see https://github.com/JetBrains/kotlin/commit/53210770a6877c5c08735070f8eff3e33573f0f5)
+ * which causes the compiler to throw: "java.lang.IncompatibleClassChangeError: Found class org.jetbrains.kotlin.ir.types.IrSimpleType, but interface was expected"
+ * when the compiler plugin is compiled with Kotlin 1.6.10.
+ */
+fun getCollectionElementType(backingFieldType: IrType): IrType? {
```

Comment:
Should we create an issue for tracking these kinds of hacks? Then we can reference them and use that as a guide when we eventually upgrade our minimum supported version of the Kotlin Compiler.